state/core_access.go

```diff
@@ -334,26 +327,21 @@ func (ca *CoreAccessor) BalanceForAddress(ctx context.Context, addr Address) (*B
 	// TODO @renaynay: once https://github.com/cosmos/cosmos-sdk/pull/12674 is merged, use this method
 	// instead
 	prefixedAccountKey := append(banktypes.CreateAccountBalancesPrefix(addr.Bytes()), []byte(app.BondDenom)...)
-	abciReq := abci.RequestQuery{
+	req := &tmservice.ABCIQueryRequest{
+		Data: prefixedAccountKey,
 		// TODO @renayay: once https://github.com/cosmos/cosmos-sdk/pull/12674 is merged, use const instead
 		Path:   fmt.Sprintf("store/%s/key", banktypes.StoreKey),
 		Height: int64(head.Height() - 1),
-		Data:   prefixedAccountKey,
 		Prove:  true,
 	}
-	opts := rpcclient.ABCIQueryOptions{
-		Height: abciReq.GetHeight(),
-		Prove:  abciReq.GetProve(),
-	}
-	result, err := ca.rpcCli.ABCIQueryWithOptions(ctx, abciReq.GetPath(), abciReq.GetData(), opts)
-	if err != nil {
+
+	result, err := ca.abciQueryCli.ABCIQuery(ctx, req)
+	if err != nil || result.Code != 0 {
```

Comment:
If we hit result.Code != 0, the err might be nil, returning both values nil, when there is an error.
Is there err field on the result struct?